generic/tests/linux_stress.py

```diff
@@ -19,19 +22,42 @@ def run(test, params, env):
     :param env: Dictionary with test environment
     """
 
-    vm = env.get_vm(params['main_vm'])
-    vm.verify_alive()
-    stress = utils_test.VMStress(vm, 'stress', params)
-    stress.load_stress_tool()
-    stress_duration = int(params.get('stress_duration', 0))
-    # NOTE: stress_duration = 0 ONLY for some legacy test cases using
-    # autotest stress.control as their sub test.
-    # Please DO define stress_duration to make sure the clean action
-    # being performed, if your case can not be controlled by time,
-    # use utils_test.VMStress() directly
+    stress_duration = int(params.get("stress_duration", "0"))
+    stress_type = params.get("stress_type", "stress")
+    vms = env.get_all_vms()
+    up_time = {}
+    error = False
+    stress_server = {}
+
+    for vm in vms:
+        try:
+            up_time[vm.name] = vm.uptime()
+            stress_server[vm.name] = utils_test.VMStress(vm, stress_type, params)
+            stress_server[vm.name].load_stress_tool()
+        except exceptions.TestError as err_msg:
+            error = True
+            logging.error(err_msg)
 
     if stress_duration:
         time.sleep(stress_duration)
-        stress.unload_stress()
-        stress.clean()
-        vm.verify_kernel_crash()
+
+    for vm in vms:
+        try:
+            uptime = vm.uptime()
```

Comment:
you can make this uptime check in try block as you suspect of below network issue, can be checked in the same block, as it will any way fail out here itself incase of network not reachable.

Reply:
Yes you are correct. Thanks!